src/main/java/org/radarcns/management/security/ClaimsTokenEnhancer.java

```diff
@@ -60,13 +61,20 @@ public OAuth2AccessToken enhance(OAuth2AccessToken accessToken,
                             + role.getAuthority().getName())
                         .collect(Collectors.toList());
                     additionalInfo.put("roles", roles);
+
+                }
+
+                Subject subject = subjectRepository.findBySubjectLogin(userName);
```

Comment:
Yes the `userName` is part of the JWT, but the `subject-id` is it's primary key. The `UserResource` endpoints are based on `login`, but the `SubjectResource` endpoints are based on `id`. So either we add the `subject-id`, or we change the endpoints in `SubjectResource` to be based on `login`. I chose not to modify the endpoints, but either solution is fine for me.

Reply:
True. But as this endpoint is specific for client registration and will not be a part of MP front-end interaction, i think we can go for `subject/{login}sources` for this specific api. so that the external parties will not get confused between `subject.getId` and `subject-id` which is the login. It would keep things simpler IMO